From: Timo Gurr <tgurr@exherbo.org>, getXconsole function from Debian
Reason: powerbtn.sh doesn't check if KDE4 session is running without the additional getXconsole function
Upstream: Author is aware of the issue, Gentoo bug http://bugs.gentoo.org/show_bug.cgi?id=253764

diff -Naur acpid-2.0.5/samples/powerbtn/powerbtn.sh acpid-2.0.5.new/samples/powerbtn/powerbtn.sh
--- acpid-2.0.5/samples/powerbtn/powerbtn.sh	2010-05-04 14:46:12.000000000 +0200
+++ acpid-2.0.5.new/samples/powerbtn/powerbtn.sh	2010-05-19 20:47:24.014230091 +0200
@@ -24,6 +24,15 @@
         export XUSER=$user
 }
 
+getXconsole() {
+        console=`fgconsole`;
+        displaynum=`ps t tty$console | sed -n -re 's,.*/X .*:([0-9]+).*,\1,p'`
+        if [ x"$displaynum" != x"" ]; then
+                export DISPLAY=":$displaynum"
+                getXuser
+        fi
+}
+
 # Skip if we are just in the middle of resuming.
 test -f /var/lock/acpisleep && exit 0
 
